api4/post.go

```diff
@@ -245,6 +246,33 @@ func updatePost(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte(rpost.ToJson()))
 }
 
+func patchPost(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequirePostId()
+	if c.Err != nil {
+		return
+	}
+
+	post := model.PostPatchFromJson(r.Body)
+
+	if post == nil {
+		c.SetInvalidParam("post")
+		return
+	}
+
+	if !app.SessionHasPermissionToChannelByPost(c.Session, c.Params.PostId, model.PERMISSION_EDIT_POST) {
+		c.SetPermissionError(model.PERMISSION_EDIT_POST)
+		return
+	}
+
```

Comment:
Hmmm looking at this permission check I think we need to change some things a bit. Can you update the permission check to be:

```
if !app.SessionHasPermissionToPost(c.Session, c.Params.PostId, model.PERMISSION_EDIT__OTHERS_POST) {
 		c.SetPermissionError(model.PERMISSION_EDIT__OTHERS_POST)
 		return
}
```

That should also let others edit a post that is not theirs if they permission but also let you edit your own posts